020-quarkus-http-non-application-endpoints/src/test/java/io/quarkus/qe/non_application/endpoint/BackwardCompatibilityTest.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.qe.non_application.endpoint;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.in;
+
+
+import java.util.Collections;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.qe.non_application.endpoint.profiles.BackwardCompatibility;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+
+@QuarkusTest
+@TestProfile(BackwardCompatibility.class)
```

Comment:
Have you considered using QuarkusProdModeTest like in https://github.com/quarkus-qe/beefy-scenarios/blob/901bd56412e26a79a0fbd7223f577f0ab372960c/019-quarkus-consul/src/test/java/io/quarkus/qe/GreetingResourceTest.java#L33 ?

Saying this because `@TestProfile` is only supported in JVM whereas using QuarkusProdModeTest supports both JVM and Native, also we can specify different properties by test case here.